keras/callbacks.py

```diff
@@ -985,23 +985,73 @@ def on_epoch_end(self, epoch, logs=None):
 
 @keras_export("keras.callbacks.TerminateOnNaN")
 class TerminateOnNaN(Callback):
-    """Callback that terminates training when a NaN loss is encountered."""
+    """Callback that terminates training when a NaN loss is encountered.
 
-    def __init__(self):
+    Args:
+        check_freq: `'batch'` or `'epoch'` or integer. When using `'epoch'`,
+            checks the losses for NaN after every epoch. If using an integer, 
+            let's say `1000`, losses will be checked every 1000 batches.
+            `'batch'` is a synonym for `1`, meaning that check will happen 
+            every batch. Defaults to `'batch'`.
+    """
+
+    def __init__(self, check_freq="batch"):
         super().__init__()
+        self.check_freq = 1 if check_freq == "batch" else check_freq
+        if self.check_freq != "epoch" and not isinstance(self.check_freq, int):
+            raise ValueError(
+                "Unrecognized check_freq: {}".format(self.check_freq)
+            )
+        self._batches_seen_since_last_saving = 0
```

Comment:
While this variable name made sense for the checkpoint callback, it should have a more appropriate name for this use case. Please rename to `_batches_seen_since_last_check`.

Reply:
renamed